Source/Core/Common/Matrix.h

```diff
@@ -0,0 +1,104 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <array>
+
+// Tiny matrix/vector library.
+// Used for things like Free-Look in the gfx backend.
+
+namespace Common
+{
+union Vec3
+{
+  Vec3() = default;
+  Vec3(float _x, float _y, float _z) : data{_x, _y, _z} {}
+
+  Vec3& operator+=(const Vec3& rhs)
+  {
+    x += rhs.x;
+    y += rhs.y;
+    z += rhs.z;
+    return *this;
+  }
+
+  std::array<float, 3> data = {};
```

Comment:
Is it a good idea to union this array with the x/y/z members? Isn't accessing the inactive member of a union considered UB?

As far as I could see, this is mostly use to read x/y/z as array; aside from the ctor initialization...so why not provide an acessor method for those cases?

Reply:
It will be fine because the underlying types are identical. https://stackoverflow.com/questions/21763002/is-it-safe-to-put-an-stdarraypod-n-in-a-union

Reply:
Interresting. Didn't know _that_, seems good to me then 👍